File: classes/java/Consumer.java
package classes.java;

public class Consumer extends Thread{

  Product product;              // produto a ser consumido

  Consumer(Product product) {
    this.product = product;     // atribui o produto para o consumidor
  }

  @Override
  public void run() { // tenta consumir dez produtos
    for(int i = 0; i < 10; i++) try {
      this.product.consume(this.getName());
      sleep((long)(Math.random() * 2000) + 1000);
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
  }

  // gera um array de consumidores
  public static Consumer[] createConsumers(Product product, int consumersNumber) {
    Consumer[] consumers = new Consumer[consumersNumber];

    for (int i = 0; i < consumers.length; i++) consumers[i] = new Consumer(product);

    return consumers;
  }
}

File: classes/python/consumer.py
from threading import Thread
from random    import random
from time      import sleep

class Consumer(Thread):

  def __init__(self, product):
    super().__init__()
    self.product = product
  
  def run(self):
    for i in range(10):
      with self.product:
        self.product.consume(self.name)
        sleep(random() * 2 + 1)
  
  @classmethod
  def create_consumers(self, product, consumers_number):
    consumers = []

    for i in range(consumers_number):
      consumers.append(Consumer(product))
    
    return consumers